res/kiwixAPI.js

```diff
@@ -0,0 +1,43 @@
+import "core-js/stable";
+import "regenerator-runtime/runtime";
+
+let zim = {toc: {}};
+
+function getSectionList(){
+    let sectionList = [];
+    for (i = 0; i < document.querySelectorAll('h1, h2, h3, h4, h5, h6').length; i++) {
+        let headerObject = document.querySelectorAll('h1, h2, h3, h4, h5, h6')[i];
+        if (headerObject.id === "") {
+            headerObject.id = "documentparserid" + i;
+        }
+        sectionList.push(headerObject);
+    }
+    return sectionList;
```

Comment:
This function modify the content of the dom in the same time it construct a list.
And the final list is equal to `document.querySelectorAll('h1, h2, h3, h4, h5, h6')`.

We can split this in two functions, one run once at document loading to add the `id`, and one just returning `document.querySelectorAll('h1, h2, h3, h4, h5, h6')` when needed.

Reply:
But then it won't remain stateless I think. I made it like that on purpose after review comments. We don't want that dependency of one run to save the state of the sections for any API method, each API method could be called independently with the use of `getSections()`

Also, what I understood from the discussions, this file should be having no control over what is happening during the load of the DOM. It is just there to serve as the API purpose once any other application uses it.

Reply:
> But then it won't remain stateless I think.

Well this is already not stateless as you are modifying the dom.

I think that what we want is that user (kiwix readers) don't need to call one method to setup and another ones to use. Just methods to use the toc and don't care about setup.
This is what is done currently (and I agree with the principle). But we don't need to (internally) do the setup all the time. We can do it only once. Splitting the code into one function doing the setup (modify the dom, maybe create the array of sections  and put it in `zim.toc` object) and other method accessing the object (internally calling the setup function if needed) is better.